File: union/frontend/src/services/poll.service.ts
import api from './api';
import { PollCreateDto, PollDeleteResponseDto, PollResponseDto, PollUpdateDto } from '../types';

export const pollService = {
  createPoll: async (unionId: number, data: PollCreateDto): Promise<PollResponseDto> => {
    const response = await api.post(`/polls/createPoll/${unionId}`, data);
    return response.data;
  },

  updatePoll: async (pollId: number, data: PollUpdateDto): Promise<PollResponseDto> => {
    const response = await api.put(`/polls/updatePoll/${pollId}`, data);
    return response.data;
  },

  getPollsByUnion: async (unionId: number): Promise<PollResponseDto[]> => {
    const response = await api.get(`/polls/listPoll/${unionId}`);
    return response.data;
  },

  vote: async (pollId: number, optionId: number): Promise<void> => {
    await api.post(`/polls/vote/${pollId}/${optionId}`);
  },

  getPollResults: async (pollId: number): Promise<PollResponseDto> => {
    const response = await api.get(`/polls/voteResults/${pollId}`);
    return response.data;
  },

  deletePoll: async (pollId: number): Promise<PollDeleteResponseDto> => {
    const response = await api.delete(`/polls/deletePoll/${pollId}`);
    return response.data;
  },
};

File: union/src/main/java/com/union/app/messaging/service/PollService.java
package com.union.app.messaging.service;

import com.union.app.messaging.dto.PollResponseDto;
import com.union.app.user.model.User;
import com.union.app.messaging.dto.PollCreateDto;
import com.union.app.messaging.dto.PollUpdateDto;
import com.union.app.messaging.model.Poll;

import java.util.List;

public interface PollService {
    Poll createPoll(Long unionId, PollCreateDto dto, User creator);

    Poll updatePoll(Long pollId, PollUpdateDto dto, User currentUser);

    void vote(Long pollId, Long optionId, User user);

    PollResponseDto getPollResults(Long pollId, User currentUser);

    void deletePoll(Long pollId, User currentUser);

    List<PollResponseDto> getPollsByUnion(Long unionId, User currentUser);
}
